examples/scala/build.sbt

```diff
@@ -23,16 +23,20 @@ val scala213 = "2.13.8"
 val deltaVersion = "1.2.0"
 
 val lookupSparkVersion: PartialFunction[String, String] = {
-  case v10x_and_above if Try {
-        v10x_and_above.split('.')(0).toInt
-      }.toOption.exists(_ >= 1) =>
+  case v21x_and_above if scala.util.Try {
```

Comment:
Can you add some plain english explanation for each case. it's getting complicated. 

I also think this can be simplified further by pulling out this version splitting into a separate function (there are multiple copies now). Also, I think the function can simply convert string `x.y.z` into a decimal number `x.y` (z will never decide anything) making the comparison far easier.

Reply:
Also, do we really need to support all previous versions? we can only support only up to last major version. In fact, just choosing spark version based on delta version will not work seamlessly as a scala versions will have to change as well if we go too far back. 

either way, I am okay with this change in this PR (add docs if you can, not a blocker), but would good to create an issue to simplify this.

Reply:
The Spark version lookup is added as part of https://github.com/delta-io/delta/commit/1118a72e0e7280b85a21c7e9f04a95cbbd427d28. The reason could be to allow running the examples on any Delta version which can be set through `env: DELTA_VERSION`.

I will simplify this